nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GetSCP.java

```diff
@@ -0,0 +1,118 @@
+@CapabilityDescription("Fetches files from an SCP Server and creates FlowFiles from them")
+@WritesAttributes({
+    @WritesAttribute(attribute = "filename", description = "The filename is set to the name of the file on the remote server"),
+    @WritesAttribute(attribute = "path", description = "The path is set to the path of the file's directory on the remote server. "
+            + "For example, if the <Remote Path> property is set to /tmp, files picked up from /tmp will have the path attribute set "
+            + "to /tmp. If the <Search Recursively> property is set to true and a file is picked up from /tmp/abc/1/2/3, then the path "
+            + "attribute will be set to /tmp/abc/1/2/3"),
+    @WritesAttribute(attribute = "file.lastModifiedTime", description = "The date and time that the source file was last modified"),
+    @WritesAttribute(attribute = "file.owner", description = "The numeric owner id of the source file"),
+    @WritesAttribute(attribute = "file.group", description = "The numeric group id of the source file"),
+    @WritesAttribute(attribute = "file.permissions", description = "The read/write/execute permissions of the source file"),
+    @WritesAttribute(attribute = "absolute.path", description = "The full/absolute path from where a file was picked up. The current 'path' "
+            + "attribute is still populated, but may be a relative path")})
+@SeeAlso(PutSCP.class)
+public class GetSCP extends GetFileTransfer {
+
+    private List<PropertyDescriptor> properties;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(SCPTransfer.HOSTNAME);
+        properties.add(SCPTransfer.PORT);
+        properties.add(SCPTransfer.USERNAME);
+        properties.add(SCPTransfer.PASSWORD);
+        properties.add(SCPTransfer.PRIVATE_KEY_PATH);
+        properties.add(SCPTransfer.PRIVATE_KEY_PASSPHRASE);
+        properties.add(SCPTransfer.REMOTE_PATH);
+        properties.add(SCPTransfer.FILE_FILTER_REGEX);
+        properties.add(SCPTransfer.PATH_FILTER_REGEX);
+        properties.add(SCPTransfer.POLLING_INTERVAL);
+        properties.add(SCPTransfer.RECURSIVE_SEARCH);
+        properties.add(SCPTransfer.IGNORE_DOTTED_FILES);
+        properties.add(SCPTransfer.DELETE_ORIGINAL);
+        properties.add(SCPTransfer.CONNECTION_TIMEOUT);
+        properties.add(SCPTransfer.DATA_TIMEOUT);
+        properties.add(SCPTransfer.HOST_KEY_FILE);
+        properties.add(SCPTransfer.MAX_SELECTS);
+        properties.add(SCPTransfer.REMOTE_POLL_BATCH_SIZE);
+        properties.add(SCPTransfer.STRICT_HOST_KEY_CHECKING);
+        properties.add(SCPTransfer.USE_KEEPALIVE_ON_TIMEOUT);
+        properties.add(SCPTransfer.USE_COMPRESSION);
+        properties.add(SCPTransfer.USE_NATURAL_ORDERING);
+        this.properties = Collections.unmodifiableList(properties);
+    }
```

Comment:
Given NIFI-1384 and NIFI-1318 I would discourage doing the above in the init method and instead use static initializer.
You can look at Kafka, AMQP as well as this outstanding PR for an example https://github.com/apache/nifi/pull/392/files